slack-message.c

```diff
@@ -499,15 +515,28 @@ void slack_user_typing(SlackAccount *sa, json_value *json) {
 		serv_got_typing(sa->gc, user->object.name, 4, PURPLE_TYPING);
 	} else if (user && (chan = (SlackChannel*)slack_object_hash_table_lookup(sa->channels, channel_id))) {
 		/* Channel */
-#if 0
 		PurpleConvChat *chat = slack_channel_get_conversation(sa, chan);
 		PurpleConvChatBuddy *cb = chat ? purple_conv_chat_cb_find(chat, user->object.name) : NULL;
 		if (cb) {
 			purple_conv_chat_user_set_flags(chat, user->object.name, cb->flags | PURPLE_CBFLAGS_TYPING);
-			/* TODO: clear typing after 4 seconds (like purple_conv_im typing_timeout), but only if no more recent update, and don't crash if we're gone:
-			 * schedule a timeout and set a cb->attribute with the current time? keep a list of current typing chat/user pairs? */
+			
+			guint timeout = GPOINTER_TO_UINT(g_dataset_get_data(user, "typing_timeout"));
```

Comment:
What was wrong with attaching this to `cb` as in the first version? That does seem more correct, as now if the same user is typing in multiple chats simultaneously (somehow) the first won't get cleared, but I'm assuming that didn't work for some reason.  With `user` we could add a list of these to the SlackUser (or even `SlackChannel`) object instead.

Reply:
For some reason the memory for cb was transient and would cause crashes. No idea why, maybe libpurple was freeing something when refreshing the room user list or something?